src/AyonCppApi/AyonCppApi.cpp

```diff
@@ -32,57 +32,228 @@
+#endif
+}
+// ------------------------------------------------
+
+
+AyonApi::AyonApi(const std::optional<std::string> &logFilePos,
                  const std::string &authKey,
                  const std::string &serverUrl,
                  const std::string &ayonProjectName,
                  const std::string &siteId,
-                 std::optional<int> concurrency):
-    m_num_threads(concurrency.value_or(std::max(int(std::thread::hardware_concurrency() / 2), 1))),
-    m_authKey(authKey),
-    m_serverUrl(serverUrl),
-    m_ayonProjectName(ayonProjectName),
-    m_siteId(siteId) {
+                 std::optional<int> concurrency)
+    : m_num_threads(concurrency.value_or(std::max(int(std::thread::hardware_concurrency() / 2), 1))),
+      m_authKey(authKey),
+      m_serverUrl(serverUrl),
+      m_ayonProjectName(ayonProjectName),
+      m_siteId(siteId) {
     PerfTimer("AyonApi::AyonApi");
 
     // ----------- Init m_Logger
-    std::filesystem::path logFileName = "logFile.json";
-    std::filesystem::path basePath = logFilePos;
-    std::filesystem::path logFilePath = std::filesystem::absolute(basePath) / logFileName;
+    std::filesystem::path logPath;
+    if (logFilePos.has_value()) {
+        std::filesystem::path inPath(logFilePos.value());
 
-    if (std::filesystem::exists(logFilePath)) {
-        logFilePath = std::filesystem::canonical(logFilePath);
-    }
-    else {
-        std::filesystem::create_directories(logFilePath.parent_path());
-    }
+        if (inPath.is_relative()) {
+            logPath = std::filesystem::weakly_canonical(inPath);
+        }
+        if (!inPath.has_parent_path()) {
+            // if the input path is just an filename we will just throw it into tmp
+            logPath = std::filesystem::temp_directory_path() / inPath;
+        }
+        // we allways want the data to be a json, so we just enforce it.
+        logPath.replace_extension(".json");
 
-    m_Log = std::make_shared<AyonLogger>(AyonLogger::getInstance(logFilePath.string()));
-    m_Log->LogLevlWarn();
+
+        if (std::filesystem::exists(logPath)) {
+            logPath = std::filesystem::canonical(logPath);
+        }
+        else {
+            std::filesystem::create_directories(logPath.parent_path());
+        }
+    }
+    m_Log = std::make_shared<AyonLogger>(AyonLogger::getInstance(logPath.string()));
+    m_Log->LogLevelWarn();
 
     m_Log->info(m_Log->key("AyonApi"), "Init AyonServer httplib::Client");
     m_AyonServer = std::make_unique<httplib::Client>(m_serverUrl);
-    m_AyonServer->set_bearer_token_auth(m_authKey);
+    std::cout << "After creating httplib::Client - " << m_serverUrl << std::endl;
+
+    if (isSSL()) {
+        try {
+            std::string opensslDirCLI = getOpenSSLDirByCLI();
+
+            #ifdef _WIN32
+            std::string certFileCLI = opensslDirCLI + "\\cert.pem";
+            #else
+            std::string certFileCLI = opensslDirCLI + "/cert.pem";
+            #endif
```

Comment:
Shouldn't we use std::filesystem::path to handle the multi-platform case here (and wherever applicable) ?

```suggestion
            std::filesystem::path opensslDirCLI(getOpenSSLDirByCLI());
            opensslDirCLI =/ "cert.pem";
            std::string certFileCLI = opensslDirCLI.string()
```

Reply:
```suggestion
            std::filesystem::path opensslDirCLI(getOpenSSLDirByCLI());
            opensslDirCLI /= "cert.pem";
            std::string certFileCLI = opensslDirCLI.string()
```